src/commands.rs

```diff
@@ -1634,257 +1697,362 @@ pub fn zremrangebylex<K: Into<RedisKey>, M: Into<String>, N: Into<String>>(inner
-    match protocol_utils::frame_to_single_result(frame)? {
-      RedisValue::Integer(i) => if i < 0 {
-        Err(RedisError::new(RedisErrorKind::ProtocolError, "Invalid ZINTERSTORE response. Expected non-negative integer."))
-      }else{
-        Ok(i as usize)
-      },
-      _ => Err(RedisError::new(RedisErrorKind::ProtocolError, "Invalid ZINTERSTORE response. Expected integer."))
-    }
-  }))
-}
-
-pub fn zunionstore<D: Into<RedisKey>, K: Into<MultipleKeys>, W: Into<MultipleWeights>>(inner: &Arc<RedisClientInner>,
-                                                                                       destination: D,
-                                                                                       keys: K,
-                                                                                       weights: W,
-                                                                                       aggregate: Option<AggregateOptions>)
-  -> Box<Future<Item=usize, Error=RedisError>>
-{
-  let destination = destination.into();
-  let keys = keys.into();
-  let weights = weights.into();
-
-  Box::new(utils::request_response(inner, move || {
-    if keys.len() == 0 {
-      return Err(RedisError::new(RedisErrorKind::InvalidArgument, "ZUNIONSTORE numkeys cannot be 0."));
-    }
-
-    let mut args = Vec::with_capacity(5 + keys.len() + weights.len());
-    args.push(destination.into());
-    args.push(RedisValue::from_usize(keys.len())?);
-
-    for key in keys.inner().into_iter() {
-      args.push(key.into());
-    }
-    if weights.len() > 0 {
-      args.push(WEIGHTS.into());
-
-      for weight in weights.inner().into_iter() {
-        args.push(utils::f64_to_redis_string(weight)?);
-      }
-    }
-
-    if let Some(aggregate) = aggregate {
-      args.push(AGGREGATE.into());
-      args.push(aggregate.to_str().into());
-    }
-
-    Ok((RedisCommandKind::Zunionstore, args))
-  }).and_then(|frame| {
-    match protocol_utils::frame_to_single_result(frame)? {
-      RedisValue::Integer(i) => if i < 0 {
-        Err(RedisError::new(RedisErrorKind::ProtocolError, "Invalid ZUNIONSTORE response. Expected non-negative integer."))
-      }else{
-        Ok(i as usize)
-      },
-      _ => Err(RedisError::new(RedisErrorKind::ProtocolError, "Invalid ZUNIONSTORE response. Expected integer."))
-    }
-  }))
-}
-
-pub fn ttl<K: Into<RedisKey>>(inner: &Arc<RedisClientInner>, key: K) -> Box<Future<Item=i64, Error=RedisError>> {
-  let key = key.into();
-
-  Box::new(utils::request_response(inner, move || {
-    Ok((RedisCommandKind::Ttl, vec![key.into()]))
-  }).and_then(|frame| {
-    match protocol_utils::frame_to_single_result(frame)? {
-      RedisValue::Integer(i) => Ok(i),
-      _ => Err(RedisError::new(RedisErrorKind::ProtocolError, "Invalid TTL response. Expected integer."))
-    }
-  }))
+pub fn zremrangebyrank<K: Into<RedisKey>>(inner: &Arc<RedisClientInner>, key: K, start: i64, stop: i64) -> Box<Future<Item= usize, Error = RedisError>> {
```

Comment:
Everything below this line, the rest of this file, looks like it's just formatting. I think some formatting got mixed up with actual code changes in the previous commit.